src/sympc/grads/grad_functions.py

```diff
@@ -351,6 +351,44 @@ def backward(ctx: Dict[str, Any], grad: MPCTensor) -> MPCTensor:
         return x ** (y - 1) * y * grad
 
 
+class GradDiv(GradFunc):
+    """The Division gradient function."""
+
+    @staticmethod
+    def forward(ctx: Dict[str, Any], x: MPCTensor, y: Any) -> MPCTensor:
+        """Perform the feedforward and compute the result for the division operation.
+
+        Args:
+            ctx (Dict[str, Any]): Context used to save information needed in the backward pass
+            x (MPCTensor): 1st operand for the division operation
+            y (Any): 2nd operand for the divison operation
+
+        Returns:
+            x / y (MPCTensor): The result of the division
+        """
+        ctx["x"] = x
+        ctx["y"] = y
+        return x / y
+
+    @staticmethod
+    def backward(ctx: Dict[str, Any], grad: MPCTensor) -> MPCTensor:
+        """Perform the backward pass for the division operation.
+
+        Args:
+            ctx (Dict[str, Any]): Context used to retrieve the information for the backward pass
+            grad (MPCTensor): The gradient that came from the child nodes
+
+        Returns:
+            (x_grad, y_grad) (Tuple[MPCTensor]): The gradients passed to the X and Y nodes.
+        """
+        x, y = ctx["x"], ctx["y"]
+
+        grad_x = grad * (x / (y * x))
+        grad_y = ((-1) * (x)) / ((y ** 2) * grad)
```

Comment:
MAYBE: Why not save in the forward the `x / y` and then here multiply by `(-1 * x * grad_cache * grad)`

Reply:
Q:We should remove the previous method?